### PR TITLE
fix: add missing types for override file, enable vm2

### PR DESCRIPTION
**Pull request by jhockett**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-api/resources/awscloudformation/overrides-resource/APIGW/override.ts.sample [hunk 1]
```diff
@@ -1,8 +1,6 @@
 // This file is used to override the REST API resource configuration
-// import { AmplifyApigwResourceTemplate } from '@aws-amplify/cli-overrides-helper';
+import { AmplifyApigwResourceTemplate } from '@aws-amplify/cli-extensibility-helper';
+
+export function override(props: AmplifyApigwResourceTemplate) {
```
Comment (akshbhu) [inline packages/amplify-category-api/resources/awscloudformation/overrides-resource/APIGW/override.ts.sample:4]:
```suggestion
export function override(resource: AmplifyApigwResourceTemplate) {
```

**Review by akshbhu — APPROVED** [review]
just one nit

File: packages/amplify-category-api/resources/awscloudformation/overrides-resource/APIGW/package.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "overrides-for-root-stack",
+  "name": "overrides-for-apigw-stack",
```
Comment (jhockett) [inline packages/amplify-category-api/resources/awscloudformation/overrides-resource/APIGW/package.json:2]:
```suggestion
  "name": "overrides",
```

File: packages/amplify-category-storage/src/__tests__/provider-utils/awscloudformation/cdk-stack-builder/ddb-stack-transform.test.ts [hunk 3]
```diff
@@ -1,14 +1,11 @@
-/* These tests, test the tranform and in rurn the cdk builder too which is used within this file */
+/* These tests test the tranform and in run the cdk builder tool which is used within this file */
```
Comment (jhockett) [inline packages/amplify-category-storage/src/__tests__/provider-utils/awscloudformation/cdk-stack-builder/ddb-stack-transform.test.ts:1]:
```suggestion
/* These tests test the DDBStackTransform and run the cdk builder tool which is used within this file */
```

**Pull request merged**